crates/oxc_linter/src/config/mod.rs

```diff
@@ -1,163 +1,119 @@
-                })
-                .collect::<std::collections::HashMap<_, _>>(),
-            Err(e) => {
-                return Err(e);
-            }
-        };
-
-        let settings = parse_settings_from_root(&file);
-
-        // `extends` provides the defaults
-        // `rules` provides the overrides
-        let rules = RULES.clone().into_iter().filter_map(|rule| {
-            // Check if the extends set is empty or contains the plugin name
-            let in_extends = extends_hm.contains(rule.plugin_name());
-
-            // Check if there's a custom rule that explicitly handles this rule
-            let (is_explicitly_handled, policy, config) =
-                if let Some((policy, config)) = roles_hm.get(&(rule.plugin_name(), rule.name())) {
-                    // Return true for handling, and also whether it's enabled or not
-                    (true, *policy, config)
-                } else {
-                    // Not explicitly handled
-                    (false, AllowWarnDeny::Allow, &None)
-                };
-
-            // The rule is included if it's in the extends set and not explicitly disabled,
-            // or if it's explicitly enabled
-            if (in_extends && !is_explicitly_handled) || policy.is_enabled() {
-                Some(rule.read_json(config.clone()))
-            } else {
-                None
-            }
-        });
-
-        Ok(Self { rules: rules.collect::<Vec<_>>(), settings })
-    }
-
-    pub fn into_rules(mut self) -> Self {
-        self.rules.sort_unstable_by_key(RuleEnum::name);
-        self
+        serde_json::from_str::<serde_json::Value>(&file).map_err(|err| {
+            let guess = mime_guess::from_path(path);
+            let err = match guess.first() {
+                // syntax error
+                Some(mime) if mime.subtype() == "json" => err.to_string(),
+                Some(_) => "only json configuration is supported".to_string(),
+                None => {
+                    format!(
+                        "{err}, if the configuration is not a json file, please use json instead."
+                    )
+                }
+            };
+            FailedToParseConfigError(vec![Error::new(FailedToParseConfigJsonError(
+                path.to_path_buf(),
+                err,
+            ))])
+            .into()
+        })
     }
 
-    pub fn get_config(self) -> (std::vec::Vec<RuleEnum>, LintSettings) {
-        (self.rules, self.settings)
+    pub fn override_rules(&self, rules_to_override: &mut FxHashSet<RuleEnum>) {
+        let mut rules_to_replace = vec![];
+        let mut rules_to_remove = vec![];
+        for rule in rules_to_override.iter() {
+            let plugin_name = rule.plugin_name();
+            let rule_name = rule.name();
+            if let Some(rule_to_configure) =
+                self.rules.iter().find(|r| r.plugin_name == plugin_name && r.rule_name == rule_name)
```

Comment:
`FxHashSet<RuleEnum>` is keyed by `rule.name()`, I think it's safe to remove the plugin_name check and change `self.rules` to `<FxHashSet</* Rule Name String, _>`?